docs/standard/serialization/binaryformatter-migration-guide/winforms-migration-guide.md

```diff
@@ -0,0 +1,84 @@
+---
+title: "BinaryFormatter Migration Guide: WinForms Migration Guide"
+description: "This guide covers the affects the deprecation and removal of BinaryFormatter from .NET has on WinForms and recommends migration steps."
+ms.date: 7/31/2024
+no-loc: [BinaryFormatter, WinForms]
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "WinForms"
+---
+
+# WinForms Migration Guide
+
+# BinaryFormatter Removal
+
+Starting in .NET 9, `BinaryFormatter` is being moved to a separate, unsupported nuget package due to its known [security risks](https://learn.microsoft.com/dotnet/standard/serialization/binaryformatter-security-guide). For more information about the risks `BinaryFormatter` poses and our decision on its removal, see [BinaryFormatter Migration Guide](./overview.md). With BinaryFormatter’s removal, it is expected that many WinForms applications will be impacted, and action will need to be taken to complete migration to .NET 9+.
+
+# How BinaryFormatter Affects WinForms
+
+WinForms itself has and still currently contains `BinaryFormatter` code internally for scenarios such as clipboard, drag/drop, and Designer ResX, however, measures have been taken to mitigate the usage of `BinaryFormatter` behind the scenes for common primitive types, including primitive arrays and primitive lists. Usage of these types in clipboard and drag/drop scenarios will continue to work without using `BinaryFormatter` and without any gesture from the developer. Primitive types include the following
+
+- `boolean`
+- `byte`
+- `char`
+- `decimal`
+- `double`
+- `int`
+- `sbyte`
+- `single`
+- `TimeSpan`
+- `DateTime`
+- `uint`
+- `string`
+- `nint`
+- `nuint`
+
+Additional types WinForms currently supports include the following
+
+- `PointF`
+- `RectangleF`
+- `Bitmap`
+- `ImageListStreamer`
+
+## Clipboard
+
+If your clipboard scenario does not involve the types above, `BinaryFormatter` would be used when [`Clipboard.SetData`](https://learn.microsoft.com/dotnet/api/system.windows.clipboard.setdata) is called with your type and when [`Clipboard.GetData`](https://learn.microsoft.com/dotnet/api/system.windows.clipboard.getdata) is called to get your type. `BinaryFormatter` will also be used if [`Clipboard.SetDataObject(object, copy: true)`](https://learn.microsoft.com/dotnet/api/system.windows.clipboard.setdataobject#system-windows-clipboard-setdataobject(system-object-system-boolean)) is called. With the `BinaryFormatter` removal, developers will now see a string about `BinaryFormatter` being removed on the clipboard when using those APIs with types that are not intrinsically handled during serialization/deserialization.
```

Comment:
These are WPF link, is this intentional?

Reply:
>With the `BinaryFormatter` removal, developers will now see a string about `BinaryFormatter` being removed on the clipboard when using those APIs with types that are not intrinsically handled during serialization/deserialization.

I think this is not clear enough. It does not say that devs will see this exception when they try to get that data, not when using SetDataObject API.

Reply:
Oops, it isn't intentional :) but this section may be moved out since it a lot of this information is likely the same for WPF. I'll be more specific about when exception will be seen.